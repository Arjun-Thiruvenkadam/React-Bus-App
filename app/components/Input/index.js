/**
 *
 * Input
 *
 */
/* eslint-disable react/prop-types */

import React from 'react';
import './input.scss';

const Input = props => {
  let inputElement = null;
  const inputClasses = ['InputElement'];

  if (props.invalid && props.shouldValidate && props.touched) {
    inputClasses.push('Invalid');
  }

  inputElement = (
    <input
      className={inputClasses.join(' ')}
      {...props.elementConfig}
      value={props.value}
      onChange={props.changed}
    />
  );

  return (
    <div className="Input">
      <label className="Label">{props.label}</label>
      {inputElement}
    </div>
  );
};

export default Input;
