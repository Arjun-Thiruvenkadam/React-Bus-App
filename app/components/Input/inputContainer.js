/**
 *
 * InputContainer
 *
 */

import styled from 'styled-components';

const InputContainer = styled.div`
  width: 100%;
  padding: 10px;
  box-sizing: border-box;
`;

export default InputContainer;
