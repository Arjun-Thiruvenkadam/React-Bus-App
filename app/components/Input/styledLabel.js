/**
 *
 * StyledLabel
 *
 */

import styled from 'styled-components';

const StyledLabel = styled.label`
  font-weight: bold;
  display: block;
  margin-bottom: 8px;
`;

export default StyledLabel;
