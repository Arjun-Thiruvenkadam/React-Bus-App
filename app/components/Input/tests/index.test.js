import React from 'react';
import { render } from 'react-testing-library';
import Input from '../index';

describe('<Input />', () => {
  it('should render and match the snapshot for name', () => {
    const config = {
      elementType: 'input',
      elementConfig: {
        type: 'text',
        placeholder: 'Name',
      },
      value: '',
      validation: {
        required: true,
        minLength: 3,
      },
      valid: false,
      touched: false,
    };
    const {
      container: { firstChild },
    } = render(
      <Input
        key={1}
        elementType={config.elementType}
        elementConfig={config.elementConfig}
        value={config.value}
        invalid={!config.valid}
        shouldValidate={config.validation}
        touched={config.touched}
        changed={() => {}}
      />,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot for password', () => {
    const config = {
      elementType: 'input',
      elementConfig: {
        type: 'password',
        placeholder: 'Password',
      },
      value: '',
      validation: {
        required: true,
        minLength: 6,
      },
      valid: false,
      touched: false,
    };
    const {
      container: { firstChild },
    } = render(
      <Input
        key={1}
        elementType={config.elementType}
        elementConfig={config.elementConfig}
        value={config.value}
        invalid={!config.valid}
        shouldValidate={config.validation}
        touched={!config.touched}
        changed={() => {}}
      />,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
