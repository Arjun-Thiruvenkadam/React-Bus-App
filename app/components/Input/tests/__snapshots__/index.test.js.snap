// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render and match the snapshot for name 1`] = `
<div
  class="Input"
>
  <label
    class="Label"
  />
  <input
    class="InputElement"
    placeholder="Name"
    type="text"
    value=""
  />
</div>
`;

exports[`<Input /> should render and match the snapshot for password 1`] = `
<div
  class="Input"
>
  <label
    class="Label"
  />
  <input
    class="InputElement Invalid"
    placeholder="Password"
    type="password"
    value=""
  />
</div>
`;
